build/template-run-mi-e2e-imds.yaml

```diff
@@ -0,0 +1,49 @@
+# Managed Identity E2E tests on VM / IMDS agents
+# ---------------------------------------------
+parameters:
+  BuildConfiguration: 'Release'
+  TargetFramework:   'net8.0'
+
+steps:
+
+# Install MAUI workloads (needed for 1ES hosted agent)
+- task: Bash@3
+  displayName: Install MAUI workloads
+  inputs:
+    targetType: inline
+    script: |
+      dotnet nuget locals all --clear
+      dotnet workload install android ios macos maui
+
+# Restore
+- task: DotNetCoreCLI@2
+  displayName: Restore E2E project
+  inputs:
+    command: restore
+    projects: tests/Microsoft.Identity.Test.E2E.NetCore/Microsoft.Identity.Test.E2E.NetCore.csproj
```

Comment:
Instead of adding a new project, why don't you use test categories instead and run only specific tests ?

https://github.com/microsoft/vstest-docs/blob/main/docs/filter.md

Reply:
made it msi.csproj